src/hardware/voodoo.cpp

```diff
@@ -4640,19 +4640,22 @@ static void setup_and_draw_triangle(voodoo_state *vs)
 	auto& tmu0 = vs->tmu[0];
 	auto& tmu1 = vs->tmu[1];
 
-	/* grab the X/Ys at least */	
-	fbi.ax = (int16_t)(fbi.svert[0].x * 16.0);
-	fbi.ay = (int16_t)(fbi.svert[0].y * 16.0);
-	fbi.bx = (int16_t)(fbi.svert[1].x * 16.0);
-	fbi.by = (int16_t)(fbi.svert[1].y * 16.0);
-	fbi.cx = (int16_t)(fbi.svert[2].x * 16.0);
-	fbi.cy = (int16_t)(fbi.svert[2].y * 16.0);
+	// Vertex references
+	const auto& vertex0 = fbi.svert[0];
+	const auto& vertex1 = fbi.svert[1];
+	const auto& vertex2 = fbi.svert[2];
+
+	/* grab the X/Ys at least */
+	fbi.ax = (int16_t)(vertex0.x * 16.0);
+	fbi.ay = (int16_t)(vertex0.y * 16.0);
+	fbi.bx = (int16_t)(vertex1.x * 16.0);
+	fbi.by = (int16_t)(vertex1.y * 16.0);
+	fbi.cx = (int16_t)(vertex2.x * 16.0);
+	fbi.cy = (int16_t)(vertex2.y * 16.0);
```

Comment:
I like how you're "rolling" these nested references up iteratively, commit by commit.